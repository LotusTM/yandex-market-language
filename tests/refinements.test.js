/* eslint-env jest */

const t = require('tcomb')
const r = require('../lib/refinements')

describe('Tcomb refinement', () => {
  describe('Absoluteurl', () => {
    it('should pass absolute url', () => {
      expect(r.Absoluteurl('http://test.com')).toMatchSnapshot()
      expect(r.Absoluteurl('https://test.com')).toMatchSnapshot()
      expect(r.Absoluteurl('//test.com')).toMatchSnapshot()
    })
    it('should error on everything except absolute url', () => {
      expect(() => r.Absoluteurl(true)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl('/url')).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl('url/')).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl('url')).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(-1)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(1)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(-1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(0)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(null)).toThrowErrorMatchingSnapshot()
      expect(() => r.Absoluteurl(undefined)).toThrowErrorMatchingSnapshot()
    })
  })

  describe('Imagepath', () => {
    it('should pass path with image file', () => {
      expect(r.Imagepath('file.jpg')).toMatchSnapshot()
      expect(r.Imagepath('file.jpeg')).toMatchSnapshot()
      expect(r.Imagepath('file.webp')).toMatchSnapshot()
      expect(r.Imagepath('file.png')).toMatchSnapshot()
      expect(r.Imagepath('file.gif')).toMatchSnapshot()
      expect(r.Imagepath('file.svg')).toMatchSnapshot()
    })
    it('should error on everything except path with image file', () => {
      expect(() => r.Imagepath(true)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath('http://test.com/imagelikepath.svg/')).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath('http://test.com')).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath('/url')).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath('url/')).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath('url')).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(-1)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(1)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(-1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(1.555)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(0)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(null)).toThrowErrorMatchingSnapshot()
      expect(() => r.Imagepath(undefined)).toThrowErrorMatchingSnapshot()
    })
  })

  describe('Maxlength', () => {
    it('should pass string with less or equal length', () => {
      expect(r.Maxlength(3)(t.String)('12')).toMatchSnapshot()
      expect(r.Maxlength(3)(t.String)('123')).toMatchSnapshot()
    })
    it('should pass array with less or equal length', () => {
      expect(r.Maxlength(3)(t.list(t.Any))([1, 2])).toMatchSnapshot()
      expect(r.Maxlength(3)(t.list(t.Any))([1, 2, 3])).toMatchSnapshot()
    })
    it('should error on not tcomb type', () => {
      expect(() => r.Maxlength(3)('nope')('12345678')).toThrowErrorMatchingSnapshot()
    })
    it('should error on not `number` max', () => {
      expect(() => r.Maxlength('nope')(t.String)('12345678')).toThrowErrorMatchingSnapshot()
    })
    it('should error on long strings', () => {
      expect(() => r.Maxlength(3)(t.String)('1234')).toThrowErrorMatchingSnapshot()
    })
    it('should error on long arrays', () => {
      expect(() => r.Maxlength(3)(t.list(t.Any))([1, 2, 3, 4])).toThrowErrorMatchingSnapshot()
    })
  })
})
