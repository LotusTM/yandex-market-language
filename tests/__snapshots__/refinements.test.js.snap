// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 1`] = `"[tcomb] Invalid value true supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 2`] = `"[tcomb] Invalid value \\"/url\\" supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 3`] = `"[tcomb] Invalid value \\"url/\\" supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 4`] = `"[tcomb] Invalid value \\"url\\" supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 5`] = `"[tcomb] Invalid value -1 supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 6`] = `"[tcomb] Invalid value 1 supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 7`] = `"[tcomb] Invalid value -1.555 supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 8`] = `"[tcomb] Invalid value 1.555 supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 9`] = `"[tcomb] Invalid value 0 supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 10`] = `"[tcomb] Invalid value null supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should error on everything except absolute url 11`] = `"[tcomb] Invalid value undefined supplied to Absolute url"`;

exports[`Tcomb refinement Absoluteurl should pass absolute url 1`] = `"http://test.com"`;

exports[`Tcomb refinement Absoluteurl should pass absolute url 2`] = `"https://test.com"`;

exports[`Tcomb refinement Absoluteurl should pass absolute url 3`] = `"//test.com"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 1`] = `"[tcomb] Invalid value true supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 2`] = `"[tcomb] Invalid value \\"http://test.com/imagelikepath.svg/\\" supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 3`] = `"[tcomb] Invalid value \\"http://test.com\\" supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 4`] = `"[tcomb] Invalid value \\"/url\\" supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 5`] = `"[tcomb] Invalid value \\"url/\\" supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 6`] = `"[tcomb] Invalid value \\"url\\" supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 7`] = `"[tcomb] Invalid value -1 supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 8`] = `"[tcomb] Invalid value 1 supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 9`] = `"[tcomb] Invalid value -1.555 supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 10`] = `"[tcomb] Invalid value 1.555 supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 11`] = `"[tcomb] Invalid value 0 supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 12`] = `"[tcomb] Invalid value null supplied to Image file"`;

exports[`Tcomb refinement Imagepath should error on everything except path with image file 13`] = `"[tcomb] Invalid value undefined supplied to Image file"`;

exports[`Tcomb refinement Imagepath should pass path with image file 1`] = `"file.jpg"`;

exports[`Tcomb refinement Imagepath should pass path with image file 2`] = `"file.jpeg"`;

exports[`Tcomb refinement Imagepath should pass path with image file 3`] = `"file.webp"`;

exports[`Tcomb refinement Imagepath should pass path with image file 4`] = `"file.png"`;

exports[`Tcomb refinement Imagepath should pass path with image file 5`] = `"file.gif"`;

exports[`Tcomb refinement Imagepath should pass path with image file 6`] = `"file.svg"`;

exports[`Tcomb refinement Maxlength should error on long arrays 1`] = `
"[tcomb] Invalid value [
  1,
  2,
  3,
  4
] supplied to Array<Any>, Maxlength 3"
`;

exports[`Tcomb refinement Maxlength should error on long strings 1`] = `"[tcomb] Invalid value \\"1234\\" supplied to String, Maxlength 3"`;

exports[`Tcomb refinement Maxlength should error on not \`number\` max 1`] = `"[tcomb] Invalid value \\"nope\\" supplied to Number"`;

exports[`Tcomb refinement Maxlength should error on not tcomb type 1`] = `"[tcomb] Invalid value \\"nope\\" supplied to Type"`;

exports[`Tcomb refinement Maxlength should pass array with less or equal length 1`] = `
Array [
  1,
  2,
]
`;

exports[`Tcomb refinement Maxlength should pass array with less or equal length 2`] = `
Array [
  1,
  2,
  3,
]
`;

exports[`Tcomb refinement Maxlength should pass string with less or equal length 1`] = `"12"`;

exports[`Tcomb refinement Maxlength should pass string with less or equal length 2`] = `"123"`;
